test/integration/app.js

```diff
@@ -151,42 +144,85 @@ describe('The results page', () => {
+    });
+    describe('postcode lookup', () => {
+      describe('invalid postcodes', () => {
+        const notFoundResponse = getSampleResponse('postcodesio-responses/404.json');
+        const invalidPostcodePassingRegex = 'LS0';
+
+        it('should lookup a syntactically valid but unknown postcode and return an error message',
+          (done) => {
+            const postcodesioScope =
+              nock('https://api.postcodes.io')
+              .get(`/outcodes/${invalidPostcodePassingRegex}`)
+              .times(1)
+              .reply(404, notFoundResponse);
+            chai.request(server)
+              .get(resultsRoute)
+              .query({ location: invalidPostcodePassingRegex })
+              .end((err, res) => {
+                checkHtmlResponse(err, res);
+                expect(res.text).to
+                  .contain(messages.invalidPostcodeMessage(invalidPostcodePassingRegex));
+                expect(postcodesioScope.isDone()).to.equal(true);
+                done();
+              });
+          });
 
-    const postcodesioScope =
-    nock('https://api.postcodes.io')
-      .get(`/outcodes/${invalidPostcodePassingRegex}`)
-      .times(1)
-      .reply(404, notFoundResponse);
+        it('should validate the postcode and return an error message', (done) => {
+          nock('https://api.postcodes.io')
+            .get(`/outcodes/${invalidPostcodePassingRegex}`)
+            .times(1)
+            .reply(404, notFoundResponse);
+          const invalidPostcode = 'invalid';
+          const errorMessage =
+            `${invalidPostcode} is not a valid postcode, please try again`;
 
-    it('should lookup a syntactically valid but unknown postcode and return an error message',
-        (done) => {
           chai.request(server)
             .get(resultsRoute)
-            .query({ location: invalidPostcodePassingRegex })
+            .query({ location: invalidPostcode })
             .end((err, res) => {
               checkHtmlResponse(err, res);
-              expect(res.text).to
-                .contain(messages.invalidPostcodeMessage(invalidPostcodePassingRegex));
-              expect(postcodesioScope.isDone()).to.equal(true);
+              expect(res.text).to.contain(errorMessage);
               done();
             });
         });
+      });
 
-    it('should validate the postcode and return an error message', (done) => {
-      const invalidPostcode = 'invalid';
-      const errorMessage =
-        `${invalidPostcode} is not a valid postcode, please try again`;
-
-      chai.request(server)
-        .get(resultsRoute)
-        .query({ location: invalidPostcode })
-        .end((err, res) => {
-          checkHtmlResponse(err, res);
-          expect(res.text).to.contain(errorMessage);
-          done();
-        });
+      it('postcode server error should return an error', (done) => {
+        const postcodesioScope =
+          nock('https://api.postcodes.io')
+          .get('/postcodes/AB123CD')
```

Comment:
Can the postcode `AB123CD` be replaced by the variable to make it clear the nock is being used for the request on line 212?